compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -122,6 +122,11 @@ public void visit(BLangSimpleVarRef varRef) {
             return;
         }
 
+        if (this.currentConstSymbol == constSymbol) {
+            dlog.error(varRef.pos, DiagnosticErrorCode.SELF_REFERENCE_VAR, constSymbol.name);
```

Comment:
I think the error message should say `constant reference` instead of `variable`. Shall we add a new error message?